src/Core/Field.h

```diff
@@ -671,6 +677,27 @@ class Field
 
     Types::Which which;
 
+    /// Field may contain a Field inside in case when Field stores Array, Tuple, Map or Object.
+    /// As the result stack overflow on destruction is possible
+    /// and to avoid it we need to count the depth and have a threshold.
+    size_t nested_field_depth = 0;
```

Comment:
But it increases the size of all Fields.

Reply:
Now it will increase the size of only Array, Map, Object and Tuple. They are already big enough.